file.go

```diff
@@ -0,0 +1,205 @@
+package cfile
+
+// TODO:
+// * explore file locking
+// * use mmap, but that would make appending/trunc a lot harder.
```

Comment:
I think mmap would detract from the goal of this project. I feel like an mmap helper lib (such as this) would be super neat though. We could easily utilize that for in-house databases that sit on disk.

Reply:
well the idea was to allow a reader mmap (so you can work on a byte slice rather than an io.Reader), that has certain performance benefits

Reply:
So it would only apply to the readers?

Reply:
Yeah for starters I wanted to do it for readers-only, doing it for writers can be rather complex.